src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,57 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact in the address book. Which also has subclasses Address, Email, and Phone. 
+ */
+
+public class Contact {
+
+	public String value;
+	private boolean isPrivate;
+
+	/**
+	 * Validates given contact.
+	 *
+	 * @throws IllegalValueException if given contact string is invalid.
+	 */
+
+	public Contact(String contact, boolean isPrivate, String messageConstraints, String validRegex) throws IllegalValueException {
+		String trimmedContact = contact.trim();
+		this.isPrivate = isPrivate;
+		if (!isValid(trimmedContact, validRegex)) {
+			throw new IllegalValueException(messageConstraints);
+		}
+		this.value = trimmedContact;
+	}
+
+	/**
+	 * Returns true if a given string is a valid contact.
+	 */
+	public static boolean isValid(String test, String validRegex) {
+		return test.matches(validRegex);
+	}
+
+	@Override
+	public String toString() {
+		return value;
+	}
+
+	@Override
+	public boolean equals(Object other) {
```

Comment:
This is fine, but note that now an Email and Address having the same value will be considered equal even if they are of different types.

You can override equals in subclass but you must you must override the hasCode() too then.